### PR TITLE
Bulk Devicestatus Delete

### PR DESCRIPTION
**Pull request by jpcunningh**
This adds a new admin tools button to delete devicestatus records older than 1 month rather than deleting the entire collection.

### CODE REVIEW

File: lib/server/devicestatus.js [hunk 1]
```diff
@@ -69,6 +70,72 @@ function storage (collection, ctx) {
...
+    var dayStr;
+    var retVal = null;
+
+    var firstErr = null,
+        numDeletes = 0,
+        totalDeletes = 0,
+        totalRecordsDeleted = 0;
+
+    var endDate = moment(date);
+    endMonth = endDate.month() + 1;
```
Comment (scottleibrand) [inline lib/server/devicestatus.js]: I'm not seeing how this stops at one month ago.  The way I'm reading it, it deletes up to the current month and day?  What am I missing?

Comment (jpcunningh) [reply lib/server/devicestatus.js]: @scottleibrand, the date argument to the function is the day to delete up to. `moment(date)` converts the date string to a moment object so if the user sent an HTTP DELETE request to `$NS_URL/api/v1/devicestatus_delete/2018-02-12`, `endDate` would be set to `02/12/2018`.

**Comment by jpcunningh** [comment]
@PieterGit, can you point me to a location I can read on how and where are the swagger files used so I can test updates to them? 

Thank you!

**Comment by PieterGit** [comment]
look here:
- https://github.com/nightscout/cgm-remote-monitor/blob/dev/swagger.json
- https://github.com/nightscout/cgm-remote-monitor/blob/dev/swagger.yaml
- https://editor.swagger.io/
- https://github.com/OAI/OpenAPI-Specification

You can create the json with the swagger editor ui based on the yaml.
The API should be available on, https://YOUR-SITE.com/api-docs.html 
but I see it doesn't work at this moment, so we need to fix that bug 🐛

**Comment by jpcunningh** [comment]
Rather than a new api, I am closing this and opening a new one that enhances the capability of the existing devicestatus api.

**Comment by PieterGit** [comment]
The user documentation for this feature can be improved. I logged issue nightscout/documentation#2 for that.

**Pull request closed without merging**